### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Limits.Shapes.Equivalence

### PR DESCRIPTION
**Pull request by joelriou**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/Shapes/Equivalence.lean [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+universe v₁ v₂ u₁ u₂
+
+variable {C : Type u₁} [Category.{v₁} C] {D : Type u₂} [Category.{v₂} D]
+
+theorem hasInitial_of_equivalence (e : D ⥤ C) [IsEquivalence e] [HasInitial C] : HasInitial D :=
+  Adjunction.hasColimitsOfShape_of_equivalence e
+#align category_theory.has_initial_of_equivalence CategoryTheory.hasInitial_of_equivalence
+
+theorem Equivalence.hasInitial_iff (e : C ≌ D) : HasInitial C ↔ HasInitial D :=
+  ⟨fun h => has_initial_of_equivalence e.inverse, fun h => has_initial_of_equivalence e.functor⟩
```
Comment (mo271) [inline Mathlib/CategoryTheory/Limits/Shapes/Equivalence.lean:35]:
```suggestion
  ⟨fun h => have : HasInitial C := h; hasInitial_of_equivalence e.inverse,
   fun h => have : HasInitial D := h; hasInitial_of_equivalence e.functor⟩
```

Comment (mo271) [reply Mathlib/CategoryTheory/Limits/Shapes/Equivalence.lean:35]: Not sure if why the extra have is needed...

File: Mathlib/CategoryTheory/Limits/Shapes/Equivalence.lean [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+#align category_theory.equivalence.has_initial_iff CategoryTheory.Equivalence.hasInitial_iff
+
+theorem hasTerminal_of_equivalence (e : D ⥤ C) [IsEquivalence e] [HasTerminal C] : HasTerminal D :=
+  Adjunction.hasLimitsOfShape_of_equivalence e
+#align category_theory.has_terminal_of_equivalence CategoryTheory.hasTerminal_of_equivalence
+
+theorem Equivalence.hasTerminal_iff (e : C ≌ D) : HasTerminal C ↔ HasTerminal D :=
+  ⟨fun h => has_terminal_of_equivalence e.inverse, fun h => has_terminal_of_equivalence e.functor⟩
```
Comment (mo271) [inline Mathlib/CategoryTheory/Limits/Shapes/Equivalence.lean:43]:
```suggestion
  ⟨fun h => have : HasTerminal C := h; hasTerminal_of_equivalence e.inverse,
   fun h => have : HasTerminal D := h; hasTerminal_of_equivalence e.functor⟩
```

File: Mathlib/CategoryTheory/Limits/Shapes/Equivalence.lean (hunk 1 above)
Comment (mo271) [reply Mathlib/CategoryTheory/Limits/Shapes/Equivalence.lean]: much better like you did it..

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**